### PR TITLE
Add libxcrypt

### PR DESCRIPTION
**Pull request by chrisburr**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by chrisburr** [comment]
@conda-forge/staged-recipes, ready for review

File: recipes/libxcrypt/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        - make
+        - pkg-config
+        - perl
+
+    test:
+      commands:
+        - test ! -f "${PREFIX}/include/crypt.h"
+        - test ! -f "${PREFIX}/lib/libcrypt${SHLIB_EXT}"
+        - test ! -f "${PREFIX}/lib/pkgconfig/libxcrypt.pc"
+        - test -f "${PREFIX}/lib/libcrypt.so.1"
```
Comment (isuruf) [inline recipes/libxcrypt/meta.yaml:59]: With libnsl, we only packaged the new ABI `libnsl.so.2` and rebuilt packages. Don't we need to do the same here?

Comment (chrisburr) [reply recipes/libxcrypt/meta.yaml:59]: I was thinking of making `libcrypt.so.1` available in a standalone package so we can patch the repodata for old versions.

Comment (chrisburr) [reply recipes/libxcrypt/meta.yaml:59]: To be clear, I mean in addition to rebuilding against `libcrypt.so.2`.

Comment (isuruf) [reply recipes/libxcrypt/meta.yaml:59]: Adding in new packages as a repodata patch was problematic last I checked as conda will generate warnings saying the environment is inconsistent.

Comment (chrisburr) [reply recipes/libxcrypt/meta.yaml:59]: Ah that's annoying, I'll have a play and follow up in https://github.com/conda-forge/perl-feedstock/issues/56. Worst case installing `libxcrypt-1` can become the documented workaround to this issue.

Do you have any thoughts on the rest of this recipe?

Comment (isuruf) [reply recipes/libxcrypt/meta.yaml:59]: Makes sense

File: recipes/libxcrypt/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    - test -f "${PREFIX}/include/crypt.h"
+    - test -f "${PREFIX}/lib/libcrypt${SHLIB_EXT}"
+    - test -f "${PREFIX}/lib/libcrypt.so.2"  # [not osx]
+    - test -f "${PREFIX}/lib/libcrypt.2.dylib"  # [osx]
+    - test -f "${PREFIX}/lib/pkgconfig/libxcrypt.pc"
+    - test ! -f "${PREFIX}/lib/libcrypt.so.1"
+    - test ! -f "${PREFIX}/lib/libcrypt.1.dylib"
+
+outputs:
+  - name: libxcrypt-1
```
Comment (isuruf) [inline recipes/libxcrypt/meta.yaml:40]:
```suggestion
  - name: libxcrypt1
```

**Review by isuruf — APPROVED** [review]

File: recipes/libxcrypt/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  script_env:
+    - OBSOLETE_API=no
+  script: $RECIPE_DIR/do_build.sh
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - make
+    - pkg-config
+    - perl
```
Comment (isuruf) [inline recipes/libxcrypt/meta.yaml:27]: This is going to create a cyclic graph.

Comment (chrisburr) [reply recipes/libxcrypt/meta.yaml:27]: Is there anything we can/should do about it?

Comment (isuruf) [reply recipes/libxcrypt/meta.yaml:27]: You could use `perl` from yum.

**Pull request merged**